exchange/bitswap/bitswap.go

```diff
@@ -52,28 +53,47 @@ var (
 func New(parent context.Context, p peer.ID, network bsnet.BitSwapNetwork,
 	bstore blockstore.Blockstore, nice bool) exchange.Interface {
 
+	// important to use provided parent context (since it may include important
+	// loggable data). It's probably not a good idea to allow bitswap to be
+	// coupled to the concerns of the IPFS daemon in this way.
+	//
+	// FIXME(btc) Now that bitswap manages itself using a process, it probably
+	// shouldn't accept a context anymore. Clients should probably use Close()
+	// exclusively. We should probably find another way to share logging data
```

Comment:
agreed. creating new contexts (which we do in various places) looses the logging information.

This lack of context awareness is a drawback in the `goprocess` lib. I can make it be context aware (i.e. make processes have a `Context() context.Context` but i dont know this is the right thing to do (it moves it back to the CtxCloser world it got out of). what would make it easier on the logging/user?